bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/SecurityApi.java

```diff
@@ -84,14 +84,16 @@ public List<HomeEvent> getHomeEvents(String homeId, @Nullable ZonedDateTime fres
             throws NetatmoException {
         List<HomeEvent> events = getEvents(PARAM_HOME_ID, homeId);
 
-        // we have to rewind to the latest event just after oldestKnown
-        HomeEvent oldestRetrieved = events.get(events.size() - 1);
-        while (freshestEventTime != null && oldestRetrieved.getTime().isAfter(freshestEventTime)) {
-            events.addAll(getEvents(PARAM_HOME_ID, homeId, PARAM_EVENT_ID, oldestRetrieved.getId()));
-            oldestRetrieved = events.get(events.size() - 1);
+        // we have to rewind to the latest event just after freshestEventTime
+        if (events.size() >= 0) {
+            HomeEvent oldestRetrieved = events.get(events.size() - 1);
```

Comment:
if `events.size()` is `0` the condition evaluates to true, resulting in a call to events.get(-1); leading to the same exception.
The condition should be: 
```suggestion
        if (events.size() > 0) {
            HomeEvent oldestRetrieved = events.get(events.size() - 1);
```

Reply:
Aren't Java arrays -1 based ;-) ?
Corrected.